src/app/practice/page.tsx

```
@@ -42,7 +42,14 @@ type Phase =
   | { kind: "error"; message: string };
 
 function PracticeContent() {
-  const certId = useMemo(() => getSelectedCertId(), []);
+  // getSelectedCertId()는 localStorage를 읽으므로 SSR에서는 항상 기본값을 반환한다.
+  // 클라이언트 첫 렌더에서 곧바로 실제 선택값을 읽으면 서버가 만든 HTML(기본값 기준)과
+  // 달라져 hydration mismatch가 나므로, 첫 렌더는 서버와 동일하게 기본값으로 시작하고
+  // 마운트 후 effect에서 실제 값으로 갱신한다.
+  const [certId, setCertId] = useState(DEFAULT_CERT_ID);
+  useEffect(() => {
+    setCertId(getSelectedCertId());
```

ESLint

[error] Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

/home/jailuser/git/src/app/practice/page.tsx:51:5
  49 |   const [certId, setCertId] = useState(DEFAULT_CERT_ID);
  50 |   useEffect(() => {
> 51 |     setCertId(getSelectedCertId());
     |     ^^^^^^^^^ Avoid calling setState() directly within an effect
  52 |   }, []);
  53 |   const questionRepository = useMemo(() => new JsonQuestionRepository(certId), [certId]);
  54 |   const searchParams = useSearchParams();

(react-hooks/set-state-in-effect)